### PR TITLE
Update metadata labels.

### LINKED ISSUE #367: Add support for metadata labels to ricecooker
Ricecooker needs to support adding the new metadata labels to ContentNodes.

The following fields should be supported:
```
    grade_levels
    resource_types
    learning_activities
    accessibility_labels
    categories
    learner_needs
```

In addition, any data should be validated before being sent to Studio, ensuring that only valid labels in `le_utils` are sent.

They are all imported from le_utils.constants.labels with the following field to module mapping:
```
    grade_levels : levels
    resource_types : resource_type
    learning_activities : learning_activities
    accessibility_labels : accessibility_categories
    categories : subjects
    learner_needs : needs
```

For more information see https://github.com/learningequality/studio/issues/3356

### PR DESCRIPTION
**Pull request by AtKristijan**
Add new labels for metadata in ricecooker and validate them with le_utils

[fixes #367](https://github.com/learningequality/ricecooker/issues/367)

### CODE REVIEW

File: ricecooker/classes/nodes.py [hunk 1]
```diff
@@ -9,14 +9,21 @@
...
 from .. import __version__
 from .. import config
 from ..exceptions import InvalidNodeException
 from .licenses import License
 
 MASTERY_MODELS = [id for id, name in exercises.MASTERY_MODELS]
 ROLES = [id for id, name in roles.choices]
+GRADE_LEVELS = [id for id, name in levels.choices]
+RESOURCE_TYPES = [id for id, name in resource_type.choices]
+ACCESSIBILITY_CATEGORIES = [id for id, name in accessibility_categories.choices]
```
Comment (rtibbles) [inline ricecooker/classes/nodes.py:23]: Each of these label types has an explicit list of the available labels predefined in the le-utils module: https://github.com/learningequality/le-utils/blob/main/le_utils/constants/labels/accessibility_categories.py#L23 so can avoid doing these list comprehensions for these.

File: ricecooker/classes/nodes.py [hunk 2]
```diff
@@ -691,6 +711,31 @@ def validate(self):
...
+            self.accessibility_labels in ACCESSIBILITY_CATEGORIES
+        ), "Assumption Failed: Accessibility label must be one of the following {}".format(ACCESSIBILITY_CATEGORIES)
+
+        assert (
+            self.categories in SUBJECTS
+        ), "Assumption Failed: Categories must be one of the following {}".format(SUBJECTS)
+
+        assert (
+            self.learner_needs in LEARNER_NEEDS
+        ), "Assumption Failed: Learner must be one of the following {}".format(LEARNER_NEEDS)
```
Comment (rtibbles) [inline ricecooker/classes/nodes.py:737]: Should be `Learner Needs`

**Review by rtibbles — CHANGES_REQUESTED** [review]
A good start - we also need to include these labels in the dict that is created to be sent to Studio.

**Comment by rtibbles** [comment]
Can you ensure that this PR is rebased to latest develop? The linting and tests should all be passing.

**Comment by rtibbles** [comment]
This PR is aimed at fixing [#367](https://github.com/learningequality/ricecooker/issues/367) - it does not fix [#3356](https://github.com/learningequality/studio/issues/3356), as that needs to be implemented on Studio to allow the sending of these new labels via the ricecooker specific endpoints.

File: ricecooker/classes/nodes.py (hunk 1 above)
Comment (AtKristijan) [reply ricecooker/classes/nodes.py]: Changed.

File: ricecooker/classes/nodes.py (hunk 2 above)
Comment (AtKristijan) [reply ricecooker/classes/nodes.py]: This is changed too

**Comment by AtKristijan** [comment]
I checked and tried to rebase but it was last one. I also did pre-commit to check if everything is okay with linting

**Comment by AtKristijan** [comment]
My local machine and github is giving different results for linting. Reorder is the problem.

File: ricecooker/classes/nodes.py [hunk 3]
```diff
@@ -691,6 +710,44 @@ def validate(self):
         assert (
             self.role in ROLES
         ), "Assumption Failed: Role must be one of the following {}".format(ROLES)
+
+        assert (
+            self.grade_levels in levels.LEVELSLIST
```
Comment (rtibbles) [inline ricecooker/classes/nodes.py:715]: Need to conditionalize these assertions:

First check `if self.grade_levels is not None:`

Then you need to check each of the values if it is a list, and see if they are in the `levels.LEVELSLIST`.

File: ricecooker/classes/nodes.py [hunk 4]
```diff
@@ -691,6 +710,44 @@ def validate(self):
...
+        ), "Assumption Failed: Grade levels must be one of the following {}".format(
+            levels.LEVELSLIST
+        )
+
+        assert (
```
Comment (rtibbles) [inline ricecooker/classes/nodes.py:720]: Same goes here and for all the other items.

File: ricecooker/classes/nodes.py [hunk 5]
```diff
@@ -691,6 +710,44 @@ def validate(self):
...
+        ), "Assumption Failed: Categories must be one of the following {}".format(
+            subjects.SUBJECTSLIST
+        )
+
+        assert (
+            self.learner_needs in needs.NEEDSLIST
+        ), "Assumption Failed: Learner needs must be one of the following {}".format(
+            needs.NEEDSLIST
+        )
+
```
Comment (rtibbles) [inline ricecooker/classes/nodes.py:750]: You also need to include all of this metadata in the data that we send to Studio - in the `to_dict` method of the TreeNode: https://github.com/learningequality/ricecooker/blob/develop/ricecooker/classes/nodes.py#L531 and also the `to_dict` method of the ContentNode: https://github.com/learningequality/ricecooker/blob/develop/ricecooker/classes/nodes.py#L706

**Review by rtibbles — CHANGES_REQUESTED** [review]
Still some updates to do here.

I have had similar issues with the import sorting - it seems to be something to do with whether modules are available locally or not when the import sorting happens, but I've not found a good fix for it. Would be good to work out if you have any ideas.

**Review by rtibbles — APPROVED** [review]

**Comment by rtibbles** [comment]
Hrm, not sure why but locally my pre-commit is now agreeing with the github action.

**Pull request merged**